cmd/pitr/main.go

```diff
@@ -23,6 +25,13 @@ func main() {
 	}
 	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGTERM, os.Interrupt)
 	defer stop()
+
+	go func() {
+		http.Handle("/metrics", promhttp.Handler())
+		http.HandleFunc("/health", healthHandler)
+		log.Fatal(http.ListenAndServe(":8080", nil))
+	}()
```

Comment:
Since we are spawning an http server, we should also ensure that it is going to shutdown gracefully. We can have a separate goroutine that is monitoring the termination signal and then call https://pkg.go.dev/net/http#Server.Shutdown.